src/libdrakvuf/os.c

```diff
@@ -333,44 +333,38 @@ bool drakvuf_get_process_by_handle(drakvuf_t drakvuf, drakvuf_trap_info_t* info,
         if (!drakvuf_get_pid_from_handle(drakvuf, info, handle, &pid))
         {
             PRINT_DEBUG("[LIBDRAKVUF] Failed to get remote process pid\n");
-            return VMI_EVENT_RESPONSE_NONE;
+            return false;
         }
 
-        pass_context_t pctx =
-        {
-            .pid = &pid,
-            .process = process,
-            .dtb = dtb
-        };
-        // Get process by pid
-        drakvuf_enumerate_processes(drakvuf, process_visitor, (void*)(&pctx));
+        return drakvuf_get_process_by_pid(drakvuf, pid, process, dtb);
     }
+
     // Self process
-    else
-    {
+    if (process)
         *process = info->attached_proc_data.base_addr;
-        *dtb     = info->regs->cr3;
-    }
-
-    if (!*process || !*dtb)
-        return false;
+    if (dtb)
+        *dtb = info->regs->cr3;
     return true;
 }
 
 bool drakvuf_get_process_by_pid(drakvuf_t drakvuf, vmi_pid_t pid, addr_t* process, addr_t* dtb)
 {
     pass_context_t pctx =
     {
-        .pid = &pid,
-        .process = process,
-        .dtb = dtb
+        .pid = pid,
     };
     // Get process by pid
     drakvuf_enumerate_processes(drakvuf, process_visitor, (void*)(&pctx));
 
-    if (!*process || !*dtb)
-        return false;
-    return true;
+    if (pctx.process && pctx.dtb)
```

Comment:
Is this if check actually needed here?

Reply:
It is equivalent to "it is found". I would replace this check to more straghtforward. Thanks.